src/XMakeBuildEngine/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -490,8 +490,18 @@ private int LaunchNode(string msbuildLocation, string commandLineArgs)
             CommunicationsUtilities.Trace("Launching node from {0}", msbuildLocation);
 
 #if RUNTIME_TYPE_NETCORE
-            string coreRunName = NativeMethodsShared.IsUnixLike ? "corerun" : "CoreRun.exe";
-            string exeName = Path.Combine(Path.GetDirectoryName(msbuildLocation), coreRunName);
+            // Try to get the process name in case of different host running.
+            string hostName, exeName;
+            using (Process currentProcess = Process.GetCurrentProcess()) hostName = currentProcess.MainModule.FileName;
+            if (string.IsNullOrEmpty(hostName))
+            {
+                hostName = NativeMethodsShared.IsUnixLike ? "corerun" : "CoreRun.exe";
```

Comment:
mm actually no, I just added it as a fallback in the case where the process name couldn't be retrieved but without having a specific case in mind. I'll remove it unless somebody thinks that this might be empty. @stephentoub is that right?